src/Makefile.test.include

```diff
@@ -1168,8 +1168,19 @@ if EMBEDDED_UNIVALUE
 endif
 
 %.cpp.test: %.cpp
-	@echo Running tests: `cat $< | grep -E "(BOOST_FIXTURE_TEST_SUITE\\(|BOOST_AUTO_TEST_SUITE\\()" | cut -d '(' -f 2 | cut -d ',' -f 1 | cut -d ')' -f 1` from $<
-	$(AM_V_at)$(TEST_BINARY) --catch_system_errors=no -l test_suite -t "`cat $< | grep -E "(BOOST_FIXTURE_TEST_SUITE\\(|BOOST_AUTO_TEST_SUITE\\()" | cut -d '(' -f 2 | cut -d ',' -f 1 | cut -d ')' -f 1`" -- DEBUG_LOG_OUT > $<.log 2>&1 || (cat $<.log && false)
+	@echo Running tests: $$(\
+		cat $< | \
+		grep -E "(BOOST_FIXTURE_TEST_SUITE\\(|BOOST_AUTO_TEST_SUITE\\()" | \
+		cut -d '(' -f 2 | cut -d ',' -f 1 | cut -d ')' -f 1) \
+		from $<
+	$(AM_V_at)$(TEST_BINARY) \
+		--catch_system_errors=no -l test_suite -t "$$(\
+			cat $< | \
+			grep -E "(BOOST_FIXTURE_TEST_SUITE\\(|BOOST_AUTO_TEST_SUITE\\()" | \
+			cut -d '(' -f 2 | cut -d ',' -f 1 | cut -d ')' -f 1\
+		)" -- DEBUG_LOG_OUT > $(abs_builddir)/$$(\
+			echo $< | grep -E -o "(wallet/test/.*\.cpp|test/.*\.cpp)" | $(SED) -e s/\.cpp/.log/\
+		) 2>&1 || (cat $<.log && false)
```

Comment:
This is broken. It's still printing the old log file (which may not exist), not the new one. Included a fix in #16228.